starlite/signature.py

```diff
@@ -90,32 +100,26 @@ def construct_exception(
 
         Parameters
         ----------
-        connection : Request | WebSocket
+        method : str
+        url : URL
         exc : ValidationError
 
         Returns
         -------
         ValidationException | InternalServerException
         """
-        client_errors = []
-        server_errors = []
+        client_errors: List["ErrorDict"] = []
+        server_errors: List["ErrorDict"] = []
+
         for error in exc.errors():
-            if error["loc"][-1] in cls.factory.dependency_name_set:
-                server_errors.append(error)
-            else:
-                client_errors.append(error)
-        method = connection.method if isinstance(connection, Request) else "websocket"
+            is_server_error = error["loc"][-1] in cls.factory.dependency_name_set
+            lst = server_errors if is_server_error else client_errors
+            lst.append(error)
```

Comment:
I find this somewhat harder to parse than the original.

Totally agree that `error["loc"][-1] in cls.factory.dependency_name_set` should be abstracted away.

Maybe we should have a utility staticmethod `is_server_error()` so then the block would become:

```python
if self.is_server_error(error):
    server_errors.append(error)
...
```
Not a big deal either way, but we don't have to avoid simple `if`/`else` statements, do we?

Reply:
You’re right. I will change it.

Reply:
Done.